### ISSUE TITLE
Calendar: Unable to target the CalendarDay 'header' button with an aria-label

### ISSUE THREAD

**Issue by Lego6245** [microsoft/fluentui#15876]
### Environment Information

- **Package version(s)**: ^7.146.0
- **Browser and OS versions**: Chromium Edge + Windows 10

### Describe the issue:

You are unable to target the CalendarDay "header" button with an aria-label. The "month" and "year" pickers both are able to be targeted easily through the `ICalendarStrings` prop, but not the day picker.

### Please provide a reproduction of the issue in a codepen:

https://fabricweb.z5.web.core.windows.net/oufr/7.149.4/#/examples/calendar (Using the `Calendar with overlaid month picker launched from a button` demo)

#### Actual behavior:

You are unable to add a aria-label to the CalendarDay's `HeaderButtonComponentType` (see https://github.com/microsoft/fluentui/blob/499bddf9cc43b09c839294cea5f066d852656b87/packages/react-date-time/src/components/Calendar/CalendarDay/CalendarDay.base.tsx#L60)

#### Expected behavior:

ICalendarStrings is expanded to include a property to target an aria-label on the CalendarDay's HeaderButtonComponentType, similar to `CalendarMonth` and `CalendarYear`

### Documentation describing expected behavior

See MAS 3.3.2 for the accessibility standard requiring buttons having aria-labels that describe their behavior.

**Comment by Lego6245** [comment]
Looks like #15074 may be a related issue, but it is not referring to the same "button" within the component.

**Comment by paulgildea** [comment]
@Lego6245 Thanks for filing this issue. We'll look into it.

@smhigley FYI for model
@lorejoh12 FYI for any known issues

@behowell Looks like they were able to identify where the issue is occurring.

**Comment by Lego6245** [comment]
Hi. This issue has staled a bit, and we are hoping to fix this accessibility issue in the near future. Will this be fixed by FluentUI folks in the short term? Or should we investigate making a fix ourselves?
